src/hooks/use-user-agent/use-user-agent.ts

```diff
@@ -0,0 +1,14 @@
+import { useContext } from 'react';
+
+import { UserAgentContext } from './user-agent-context';
+
+export const useUserAgent = () => {
+  const userAgent = useContext(UserAgentContext);
+  const isMobileAgent = /iPhone|iPod|Android/i.test(userAgent);
+
+  if (userAgent === undefined) {
+    throw new Error('useUserAgent should be used within CounterProvider');
+  }
```

Comment:
```suggestion
  if (userAgent === undefined) {
    throw new Error('useUserAgent should be used within UserAgentContext.Provider');
  }
```

텍스트만 변경하면 좋을 거 같아요 !!

Reply:
복붙한거 시원하게 걸려버렸네